src/tools/compiletest/src/load_cfg.rs

```diff
@@ -0,0 +1,169 @@
+#![allow(unused)]
+
+mod itemlist;
+mod prepare;
+
+use error::{Error, ErrorExt, Result};
+
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
+pub struct LineCol {
+    pub line: usize,
+    pub col: usize,
+}
+
+impl LineCol {
+    const fn new(line: usize, col: usize) -> Self {
+        Self { line, col }
+    }
+}
+
+/// When we need to e.g. build regexes that include a pattern, we need to know what kind of
+/// comments to use. Usually we just build a regex for all expressions, even though we don't
+/// use them.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+enum CommentTy {
+    Slashes,
+    Hash,
+    Semi,
+}
+
+impl CommentTy {
+    const fn as_str(self) -> &'static str {
+        match self {
+            CommentTy::Slashes => "//",
+            CommentTy::Hash => "#",
+            CommentTy::Semi => ";",
+        }
+    }
+
+    const fn directive(self) -> &'static str {
+        match self {
+            CommentTy::Slashes => "//@",
+            CommentTy::Hash => "#@",
```

Comment:
Remark: unfortunately, `Makefile`s are still using the same directive prefix `#` as comments `#`, because I didn't try to port them when I ported `//` -> `//@` since we're porting `Makefile`s into `rmake.rs` *anyway*.

Reply:
That is an interesting twist - I think it would be easy enough to just ignore unknown directives for specific kinds of tests rather than raising errors. Unless https://github.com/rust-lang/rust/issues/121876 happens to be done sooner than expected.

Reply:
Yes, that makes sense to me. We don't really need to error for *existing* Makefile tests, and at least we are guarding the test suite with a tidy check to disallow *new* Makefiles to make it in.